### PR TITLE
Suspend Texter (part deux)

### LINKED ISSUE #1487: Feature Request: Allow deletion of a user from the People page
**Problem**
Admins currently can't delete a user. Regularly, a user will accidentally end up with 2 accounts and then the admin needs to go into the DB to delete the account that shouldn't exist.

**Solution**
Let's add a red delete button to the edit modal that appears once you hit edit on the people page. I'm a little unsure on the best positioning here.

**The edit button that opens the modal**
Screen Shot 2020-04-16 at 6 46 06 PM

**Potential place for a delete button in the modal**
Screen Shot 2020-04-16 at 6 47 15 PM

### PR DESCRIPTION
**Pull request by JeremyParker**
# Fixes #1487

## Description
This change surfaces the ability to restrict a texter to the SUSPENDED role.
- on the People screen admins can set people's role to SUSPENDED.
- suspended texters are omitted by default. Admins can filter by SUSPENDED state to see them.

- in the Message Review screen `(SUSPENDED)` appears after the suspended texters' display names.
- autofills on Message Review screen allow filtering by suspended texters, but not assigning to suspended texters.

- Campaign screen shows all active texters regardless of suspended state
- Campaign screen and Campaign Edit show `(SUSPENDED)` next to suspended texters' names.
- Campaign edit screen autofill doesn't allow adding suspended texters to the campaign.

This PR replaces https://github.com/MoveOnOrg/Spoke/pull/1562

Suspended users still appear in the Campaign Texter Stats, but with the word `(Suspended)` next to them.
Screen Shot 2020-06-26 at 4 34 05 PM

In the Campaign Edit view the suspended texter also appears with `(Suspended)` after their name.
Screen Shot 2020-06-26 at 4 34 18 PM

An admin can filter the People list by `Suspended` to see all the suspended texters. Otherwise they are not shown in the `All Roles` filter.
Screen Shot 2020-06-26 at 4 34 39 PM

In Message Review, conversations assigned to a suspended texter are still shown, with `(Suspended)` next to the texter's name.
Suspended texters still appear in the Message Filter `Texter` field autofill options.
Suspended texters do NOT appear in the `Reassign To...` field autofill.
Screen Shot 2020-06-26 at 4 36 39 PM

# Checklist:

- [x] I have manually tested my changes on desktop and mobile
- [x] The test suite passes locally with my changes
- [x] If my change is a UI change, I have attached a screenshot to the description section of this pull request
- [x] [My change is 300 lines of code or less](https://github.com/MoveOnOrg/Spoke/blob/main/CONTRIBUTING.md#your-first-code-contribution), or has a documented reason in the description why it’s longer
- [x] I have made any necessary changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] My PR is labeled [WIP] if it is in progress

### CODE REVIEW

**Comment by schuyler1d** [comment]
changed base in preparation for 7.0 merge/release

**Review by schuyler1d — DISMISSED** [review]
This was really concise and elegant.
Using this diff for review:
https://github.com/MoveOnOrg/Spoke/compare/9362b01a3987aa161cb32eec1df4f4bc154a672d...JeremyParker:suspend-user-step-2
:+1:

File: src/components/CampaignTextersForm.jsx [hunk 1]
```diff
@@ -274,6 +275,7 @@ export default class CampaignTextersForm extends React.Component {
 
     const dataSource = orgTexters
       .filter(orgTexter => !texters.find(texter => texter.id === orgTexter.id))
+      .filter(orgTexter => getHighestRole(orgTexter.roles) === "SUSPENDED")
```
Comment (schuyler1d) [inline src/components/CampaignTextersForm.jsx]: I think this needs to be `!==` though

Comment (JeremyParker) [reply src/components/CampaignTextersForm.jsx]: Good eye! I was testing this yesterday and was confused why it wasn't working. Went looking elsewhere and decided to come back to it today. ✅  problem solved 😄 
Fixed in https://github.com/MoveOnOrg/Spoke/pull/1633/commits/2eeb0c9ecc11b634a20eaea35a92fcd960c860e2

**Comment by schuyler1d** [comment]
Also seeing an issue where if I added someone to be a texter of a campaign, and then I suspend them, then that assignment (in Campaign Edit when revisited) doesn't get shown at all.  I think it should show the assignment and say "(Suspended)" next to their name instead, ideally -- and it shouldn't do anything to the assignment.

**Comment by JeremyParker** [comment]
@schuyler1d I'm not seeing this. Do you mean the Campaign summary like http://localhost:3000/admin/1/campaigns/2 on a started Campaign? That screen should probably show `(Suspended)`. I'll add that now.

**Comment by JeremyParker** [comment]
Suspended users still appear in the Campaign Texter Stats, but with the word `(Suspended)` next to them.
Screen Shot 2020-06-26 at 4 34 05 PM

**Comment by JeremyParker** [comment]
In the Campaign Edit view the suspended texter also appears with `(Suspended)` after their name.
Screen Shot 2020-06-26 at 4 34 18 PM

**Comment by JeremyParker** [comment]
An admin can filter the People list by `Suspended` to see all the suspended texters. Otherwise they are not shown in the `All Roles` filter.
Screen Shot 2020-06-26 at 4 34 39 PM

**Comment by joemcl** [comment]
Well yeah, that should be the case. You can reassign their stuff to others with Message Review.

**Comment by JeremyParker** [comment]
In Message Review, conversations assigned to a suspended texter are still shown, with `(Suspended)` next to the texter's name.
Suspended texters still appear in the Message Filter `Texter` field autofill options.
Suspended texters do NOT appear in the `Reassign To...` field autofill.
Screen Shot 2020-06-26 at 4 36 39 PM

**Comment by JeremyParker** [comment]
> Well yeah, that should be the case. You can reassign their stuff to others with Message Review.

Sorry, I just moved all these comments to the description field. This is my first time following the PR Template Checklist.

**Pull request merged**